extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php

```diff
@@ -48,65 +25,79 @@ public static function getTemplates() {
-		wfProfileIn( __METHOD__ );
-
-		$helper = new PortableInfoboxRenderServiceHelper();
-		$infoboxHtmlContent = '';
-		$heroData = [ ];
-
-		// decide on image width
-		$this->imagesWidth = $helper->isMobile() ?
-			PortableInfoboxRenderServiceHelper::MOBILE_THUMBNAIL_WIDTH :
-			// if europa go with bigger images! else default size
-			$helper->isEuropaTheme() ? PortableInfoboxRenderServiceHelper::EUROPA_THUMBNAIL_WIDTH :
-				PortableInfoboxRenderServiceHelper::DEFAULT_DESKTOP_THUMBNAIL_WIDTH;
-
-		foreach ( $infoboxdata as $item ) {
-			$data = $item[ 'data' ];
-			$type = $item[ 'type' ];
-
-			switch ( $type ) {
-				case 'group':
-					$infoboxHtmlContent .= $this->renderGroup( $data );
-					break;
-				case 'navigation':
-					$infoboxHtmlContent .= $this->renderItem( 'navigation', $data );
-					break;
-				default:
-					if ( $helper->isMobile() && $helper->isValidHeroDataItem( $item, $heroData ) ) {
-						$heroData[ $type ] = $data;
-						continue;
-					}
-
-					if ( $helper->isTypeSupportedInTemplates( $type, self::getTemplates() ) ) {
-						$infoboxHtmlContent .= $this->renderItem( $type, $data );
-					};
-			}
-		}
+	public function renderInfobox( array $infoboxdata, $theme, $layout, $accentColor, $accentColorText ) {
+		$this->inlineStyles = $this->getInlineStyles( $accentColor, $accentColorText );
 
-		if ( !empty( $heroData ) ) {
-			$infoboxHtmlContent = $this->renderInfoboxHero( $heroData ) . $infoboxHtmlContent;
-		}
+		// decide on image width, if europa go with bigger images! else default size
+		$this->imagesWidth = $this->isEuropaTheme() ? self::EUROPA_THUMBNAIL_WIDTH :
+			self::DEFAULT_DESKTOP_THUMBNAIL_WIDTH;
+
+		$infoboxHtmlContent = $this->renderChildren( $infoboxdata );
 
 		if ( !empty( $infoboxHtmlContent ) ) {
 			$output = $this->renderItem( 'wrapper', [
 				'content' => $infoboxHtmlContent,
 				'theme' => $theme,
 				'layout' => $layout,
-				'isEuropaEnabled' => $helper->isEuropaTheme()
+				'isEuropaEnabled' => $this->isEuropaTheme()
 			] );
 		} else {
 			$output = '';
 		}
 
-		\Wikia\PortableInfobox\Helpers\PortableInfoboxDataBag::getInstance()->setFirstInfoboxAlredyRendered( true );
+		return $output;
+	}
 
-		wfProfileOut( __METHOD__ );
+	protected function getImageHelper() {
+		if ( !isset( $this->helper ) ) {
+			$this->helper = new PortableInfoboxImagesHelper();
+		}
+		return $this->helper;
+	}
 
-		return $output;
+	/**
+	 * Produces HTML output for item type and data
+	 *
+	 * @param $type
+	 * @param array $data
+	 * @return string
+	 */
+	protected function render( $type, array $data ) {
+		return $this->templateEngine->render( $type, $data );
+	}
+
+	/**
+	 * renders part of infobox
+	 *
+	 * @param string $type
+	 * @param array $data
+	 *
+	 * @return string - HTML
+	 */
+	protected function renderItem( $type, array $data ) {
+		switch ( $type ) {
+			case 'group':
+				$result = $this->renderGroup( $data );
+				break;
+			case 'header':
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "header" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)